src/services/identity/QueueClient.ts

```diff
@@ -0,0 +1,91 @@
+import AWS, { SQS } from "aws-sdk"
+
+import logger from "@root/logger/logger"
+
+const { INCOMING_QUEUE_URL, OUTGOING_QUEUE_URL } = process.env
+export default class QueueClient {
+  private readonly sqs: AWS.SQS
+
+  private readonly incomingQueueUrl: string
+
+  private readonly outgoingQueueUrl: string
+
+  constructor() {
+    this.sqs = new AWS.SQS()
+    if (!INCOMING_QUEUE_URL || !OUTGOING_QUEUE_URL) {
+      throw Error(`Queue URLs are not configured in environment variable`)
+    }
+    this.incomingQueueUrl = INCOMING_QUEUE_URL
+    this.outgoingQueueUrl = OUTGOING_QUEUE_URL
+  }
+
+  sendMessage = async (MessageBody: string) => {
+    const queueResestParams: SQS.Types.SendMessageRequest = {
+      QueueUrl: this.outgoingQueueUrl,
+      MessageBody,
+    }
+    this.sqs.sendMessage(queueResestParams, (err, data) => {
+      if (err) {
+        logger.error(err)
+      } else {
+        logger.info(data)
+      }
+    })
+  }
+
+  receiveMessage = async () => {
+    logger.info(`checking queue`)
+    const params: SQS.ReceiveMessageRequest = {
+      QueueUrl: this.incomingQueueUrl,
+      AttributeNames: ["All"],
+      MaxNumberOfMessages: 10,
+    }
+
+    /**
+     * Note: using `.promise` might be an issue. See more: https://github.com/aws/aws-sdk-js/issues/1453
+     * Through some internal testing, this issue "seems" to have disappeared (it is an undeterministic bug), assumed to
+     * be a safe operation for now.
+     */
+    const response = await this.sqs.receiveMessage(params).promise()
+    if (response.$response.error) {
+      logger.error(response.$response.error)
+    }
```

Comment:
Hmm i rather not here. Reason being: if there is an error thrown here, it shows the existance of an error from our backend _communicating_ with the queue service. This gives us no impression to whether or not the domain association process was indeed a success/failure. My hypothesis is that this will most likely fail due to network blips, and I rather not send an email to anyone unless I know for sure we have an _**error that we cannot recover from**_. In the case of network blips, the messages will still reside in the queue (since they are not deleted yet), and we can still recevie them in the next polling method call. wdty?

Reply:
your reasoning makes sense! so to recap, since this is just a communication failure between our queue and be, the most likely reason is the network being faulty (as opposed to, for eg, our be being down). as the messages have not been deleted, they would still be there in the queue and the next poll should read them assuming good network conditions